src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -773,3 +774,20 @@ export const selectPortfolioBridgeAssets = createDeepEqualOutputSelector(
     })
   },
 )
+
+export const selectAccountIdByAccountNumberAndChainId = createSelector(
+  selectWalletAccountIds,
+  selectPortfolioAccountMetadata,
+  selectAccountNumberParamFromFilter,
+  selectChainIdParamFromFilter,
+  (walletAccountIds, accountMetadata, accountNumber, chainId): AccountId | undefined => {
+    if (!isValidAccountNumber(accountNumber)) throw new Error('invalid account number')
+    if (chainId === undefined || !isChainId(chainId)) throw new Error('invalid chain id')
```

Comment:
Curious about the right side of this logical or, `selectChainIdParamFromFilter` should already ensure we're passing a `ChainId` (or undefined, which we're guarding against on the left side) at type-level?